src/app/core/components/two-factor-auth/two-factor-auth.component.spec.ts

```diff
@@ -0,0 +1,121 @@
+/* @format */
+import { Shallow } from 'shallow-render';
+
+import { CoreModule } from '@core/core.module';
+import { MessageService } from '@shared/services/message/message.service';
+import { AbstractControl } from '@angular/forms';
+import { TwoFactorAuthComponent } from './two-factor-auth.component';
+
+describe('TwoFactorAuthComponent', () => {
+  let shallow: Shallow<TwoFactorAuthComponent>;
+  let messageShown: boolean = false;
+
+  beforeEach(async () => {
+    messageShown = false;
+    shallow = new Shallow(TwoFactorAuthComponent, CoreModule).mock(
+      MessageService,
+      {
+        showError: () => {
+          messageShown = true;
+        },
+      }
+    );
+
+    it('should create', async () => {
```

Comment:
It seems like you didn't close the `beforeEach` block properly. Can you fix this?